src/descriptor/tr.rs

```diff
@@ -263,7 +286,7 @@ impl<Pk: MiniscriptKey, Ext: Extension> Tr<Pk, Ext> {
     /// Checks whether the descriptor is safe.
     pub fn sanity_check(&self) -> Result<(), Error> {
         for (_depth, ms) in self.iter_scripts() {
-            ms.sanity_check()?;
+            ms.as_miniscript().unwrap().sanity_check()?;
```

Comment:
In 899269acefeb534f140cfd4d7f86b66ad567b55f:

I'd be tempted to just return `false` here rather than panicking. Though I'd leave a FIXME comment.

Reply:
Ah, never mind, this is replaced in a later commit (and has a TODO comment on it there to implement it more fully).